omod-2.0/src/main/java/org/openmrs/module/attachments/rest/AttachmentResource2_0.java

```diff
@@ -1,17 +1,168 @@
+ */
 @Resource(name = RestConstants.VERSION_1 + "/attachment", supportedClass = Attachment.class, supportedOpenmrsVersions = {
         "2.0.0" })
 public class AttachmentResource2_0 extends AttachmentResource1_10 {
 	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#getRepresentationDescription(org.openmrs.module.webservices.rest.web.representation.Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("dateTime");
+			description.addProperty("comment");
+			description.addProperty("complexData");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("dateTime");
+			description.addProperty("comment");
+			description.addProperty("auditInfo");
+			description.addProperty("complexData");
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("comment");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getCreatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty("uuid");
+		description.addRequiredProperty("display");
+		description.addProperty("comment");
+		description.addProperty("dateTime");
+		description.addProperty("complexData");
+		
+		return description;
+	}
+	
+	/**
+	 * @throws org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#getUpdatableProperties()
+	 */
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("gender");
+		description.addProperty("display");
+		description.addProperty("comment");
+		description.addProperty("dateTime");
+		description.addRequiredProperty("complexData");
+		return description;
+	}
+	
+	@Override
+	public Model getGETModel(Representation rep) {
```

Comment:
can you create test units for this GET method especially for newly added properties ??

Reply:
All these methods are overridden @mozzy, hope you saw that

Reply:
@sherrif10 , does that mean we cant write unit tests for them ??

Reply:
No worry am going to write them